### PR TITLE
#TP-38 ft:update-profile

### PR DESCRIPTION
**Pull request by jeandamascyifuzo**
**What this PR does**
This pr gives the operator or driver the ability to update their profile

- [x] The operator should be able to update information about his/her own profile.
- [x] The driver should be able to update information about his/her own profile.
- [x] The operator and driver profile should be able accessible publicly.
- [x] Appropriate error messages are displayed on unsuccessful attempts.

**Trello card link**
[update profile](https://trello.com/c/UJYesI3N/38-update-profile)

**How you can manually test it**
Clone the repo
then checkout to ft-update-profile-TP-38 branch

**Installation of dependencies.**
npm i

**To update your DB run this**

npm run migrate
npm run seedAll

### CODE REVIEW

File: controllers/profileContoller.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { users, profile } from "../models";
+import { success, fail, sendError } from "../function/respond.js";
+
+
+const createProfile = async (req, res) => {
+	try {
+
+		/* =============================== start: Validation ============================== */
+		if (!req.body.profileImage) throw new Error('profile image link required');
```
Comment (Jkadhuwa) [inline controllers/profileContoller.js]: Why redefine the wheel, when we already have joi installed.

File: controllers/profileContoller.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		const updatedProfile = await users.findOne({ where: { id: profile_id } })
+
+		updatedProfile.profileImage = profileImage
+
+		await updatedProfile.save()
+
+		return success(res, 200, { updatedProfile: profileImage }, "updateProfile", req);
+
+	} catch (err) {
+		console.log(err)
```
Comment (Jkadhuwa) [inline controllers/profileContoller.js]: - No need of this console (Remove all unused consoles)
- Use the custom response handler sendError() {consistency}

File: controllers/userController.js [hunk 3]
```diff
@@ -26,14 +26,14 @@ const createUser = async (req, res) => {
     try {
 
 	    /* =============================== start: Validation ============================== */ 
-			if(!req.body.username && !req.body.password &&  !req.body.fullname && !req.body.email) throw new Error('Body is required');				
+			if(!req.body.username && !req.body.password &&  !req.body.fullname && !req.body.email && !req.body.profileImage) throw new Error('Body is required');
```
Comment (Jkadhuwa) [inline controllers/userController.js:18]: Also here use joi to validate your body, in the routes section before hitting the controller.

File: test/user.test.js [hunk 4]
```diff
@@ -22,10 +22,60 @@ describe('Test one : users', () => {
...
 	});
+
+/*============= user profile test ============*/
+
+	it("should create a User's profile", async () => {
+		const user = {
+			profileImage: 'https://cdn.pixabay.com/photo/2018/03/15/23/16/car-3229816__480.jpg',
+		};
+		const response = await chai.request(app).post(`/api/v1/profile/register`).send(user);
+		expect(response).to.have.status(201);
```
Comment (Jkadhuwa) [inline test/user.test.js]: Add more assertions to your tests, you can check the response {message, etc}

File: test/user.test.js [hunk 5]
```diff
@@ -22,10 +22,60 @@ describe('Test one : users', () => {
...
+		const response = await chai.request(app).post(`/api/v1/profile/register`).send(user);
+		expect(response).to.have.status(500);
+	});
+	// it('should update a route', async () => {
+	// 	const id = '1'
+	//     const oneRoute = {
+    //         name: "Karuruma - Kanombe",
+    //         code: "507",
+    //         city: "Rwamagana",
+    //         startLocation: "Rwamagan",
```
Comment (Jkadhuwa) [inline test/user.test.js]: Remove all commented / unused code. (unnecessary noise)

**Review by Jkadhuwa — CHANGES_REQUESTED** [review]
Nice work @jeandamascyifuzo, act on the feedback below.

File: models/profile.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+const {
+  Model
+} = require('sequelize');
+module.exports = (sequelize, DataTypes) => {
+  class Profile extends Model {
```
Comment (chrisostome01) [inline models/profile.js]: I don't think we need this model since this column is located in in users model, you delete it I guess

**Review by chrisostome01 — CHANGES_REQUESTED** [review]
Well done @jeandamascyifuzo , take look on this model

File: function/validation.js [hunk 7]
```diff
@@ -67,4 +67,13 @@ export const  validatePermissionAssignment  = (data) => {
...
 }
-/* ========= End: Assign Permissions validation ============== */ 
\ No newline at end of file
+/* ========= End: Assign Permissions validation ============== */ 
+
+/* ========= Start: update profile  validation ============== */ 
+export const  validateProfile  = (data) => {
+    /* c8 ignore next 4 */
+    const schema = Joi.object({id: Joi.number().required(),profileImage: Joi.string().required()});
+    const value = schema.validate(data , { abortEarly: false });
```
Comment (chanceDira) [inline function/validation.js:76]: @jeandamascyifuzo great work on this. let keep using joi validation.

**Review by chanceDira — APPROVED** [review]
**_Well done._**

**Review by chrisostome01 — APPROVED** [review]

**Review by Jkadhuwa — APPROVED** [review]

**Pull request merged**